Upgrading Hibernate validator from 7 to 9

gauthamkrishnanibm opened the pull request:
Upgrading Hibernate Validator from 7.0.4.Final to 9.0.0.Beta3

gauthamkrishnanibm committed:
Initial commit for upgrading Hibernate validator to 9

gauthamkrishnanibm committed:
More changes for upgrading hibernate validator

gauthamkrishnanibm committed:
changes in com.ibm.ws.beanvalidation.v20.cdi project

gauthamkrishnanibm committed:
Hibernate Validator 9.0+ package location for InjectingConstraintValidatorFactory dependency correction

gauthamkrishnanibm commented:
#build #libby

gauthamkrishnanibm commented:
#build

gauthamkrishnanibm commented:
fixes #29606

gauthamkrishnanibm committed:
update dev/com.ibm.ws.beanvalidation.v20.cdi/bnd.bnd

gauthamkrishnanibm commented:
#build

gauthamkrishnanibm commented:
#libby

gauthamkrishnanibm committed:
update org.hibernate.validator.cdi.spi version in bnd

gauthamkrishnanibm commented:
#build #libby

mswatosh commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.1.feature:
```diff
@@ -0,0 +1,14 @@
+visibility=private
+IBM-App-ForceRestart: install, uninstall
+IBM-Provision-Capability: \
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-3.0)(osgi.identity=io.openliberty.cdi-4.0)(osgi.identity=io.openliberty.cdi-4.1)))", \
```
```suggestion
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-4.1)))", \
```
Since cdi-4.1 is the EE11 version, I don't believe there's value to including 3.0 and 4.0

mswatosh commented on dev/com.ibm.ws.beanvalidation.v20.cdi/bnd.bnd:
Copyright date needs to be updated

mswatosh commented on dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/BvalManagedObjectBuilderImpl.java:
```diff
@@ -1,10 +1,10 @@
 /*******************************************************************************
- * Copyright (c) 2017, 2018 IBM Corporation and others.
+ * Copyright (c) 2017, 2018, 2024 IBM Corporation and others.
```
```suggestion
 * Copyright (c) 2017, 2024 IBM Corporation and others.
```
Copyright years are just first and latest update so 2018 should be removed

mswatosh commented on dev/io.openliberty.org.hibernate.validator.9.0/bnd.overrides:
```diff
@@ -0,0 +1,79 @@
+
+Bundle-Name: Hibernate Validator Engine
+Bundle-SymbolicName: io.openliberty.org.hibernate.validator.9.0
+Bundle-Description: Hibernate's Jakarta Bean Validation 3.1 reference implementation; version=9.0.0.Beta3
```
```suggestion
Bundle-Description: Hibernate's Jakarta Validation 3.1 reference implementation; version=9.0.0.Beta3
```

mswatosh commented on dev/io.openliberty.org.hibernate.validator.9.0/bnd.overrides:
```diff
@@ -0,0 +1,79 @@
+Bundle-SymbolicName: io.openliberty.org.hibernate.validator.9.0
+Bundle-Description: Hibernate's Jakarta Bean Validation 3.1 reference implementation; version=9.0.0.Beta3
+
+# Provide these as resources to applications
```
```suggestion
# Provide these as resources to applications
#TODO add the 3.1 schemas when upgrading to the final Hibernate Validator 9.0
```

mswatosh commented on dev/io.openliberty.org.hibernate.validator.9.0/bnd.bnd:
```diff
@@ -0,0 +1,23 @@
+	com.ibm.ws.junit.extensions;version=latest, \
+	org.hamcrest:hamcrest-all;version=1.3, \
+	org.jmock:jmock-junit4;strategy=exact;version=2.5.1, \
+	org.jmock:jmock;strategy=exact;version=2.5.1
```
```suggestion
-testpath: \
	../build.sharedResources/lib/junit/old/junit.jar;version=file, \
	com.ibm.ws.junit.extensions;version=latest
```
These extra test dependencies don't seem necessary anymore, the unit test passes without them.

mswatosh commented on dev/io.openliberty.org.hibernate.validator.9.0/bnd.overrides:
```diff
@@ -0,0 +1,79 @@
+  javax.xml.validation;version=!, \
+  jakarta.el;version="[5.0.0,7.0.0)";resolution:=optional, \
+  org.xml.sax;version=!, \
+  org.jboss.logging;version="[3.1.0,4.0.0)", \
+  com.fasterxml.classmate;version="[1.3,2.0.0)", \
+  com.fasterxml.classmate.members;version="[1.3,2.0.0)", \
+  org.joda.time;version="[2.0.0,3.0.0)";resolution:=optional, \
+  org.jsoup;version="[1.5.2,2.0.0)";resolution:=optional, \
+  org.jsoup.nodes;version="[1.5.2,2.0.0)";resolution:=optional, \
+  org.jsoup.parser;version="[1.5.2,2.0.0)";resolution:=optional, \
+  org.jsoup.safety;version="[1.5.2,2.0.0)";resolution:=optional, \
+  javax.money;version="[1.0.0,2.0.0)";resolution:=optional, \
+  com.thoughtworks.paranamer;version="[2.5.5,3.0.0)";resolution:=optional, \
```
Versions on these should be checked, and some may be removable. From what I can see HV no longer depends on `org.jsoup`

https://mvnrepository.com/artifact/org.hibernate.validator/hibernate-validator/9.0.0.Beta3

mswatosh commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee10.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen.xmlschema=[3.0,4)
 org.glassfish.jaxb.runtime.v2.util=[3.0,4)
-org.hibernate.validator=[7.0,8)
```
I'm not sure we actually need these. I believe HV8 switched from the javax to jakarta namespace, so I don't know that we need to run the transformer. We should try without these changes to confirm

mswatosh commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee10plus.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen.xmlschema=[3.0,4)
 org.glassfish.jaxb.runtime.v2.util=[3.0,4)
-org.hibernate.validator=[7.0,8)
```
Same comment as `jakarta-versions-ee10.properties`

mswatosh commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee11.properties:
```diff
@@ -217,18 +217,19 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen.xmlschema=[3.0,4)
 org.glassfish.jaxb.runtime.v2.util=[3.0,4)
-org.hibernate.validator=[7.0,8)
```
Same comment as jakarta-versions-ee10.properties

mswatosh replied:
Actually we may be able to remove these altogether for the ee11 version, since HV9 should be using the jakarta namespace

mswatosh commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen.xmlschema=[3.0,4)
 org.glassfish.jaxb.runtime.v2.util=[3.0,4)
-org.hibernate.validator=[7.0,8)
```
Same comment as jakarta-versions-ee10.properties

mswatosh commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9and10.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen.xmlschema=[3.0,4)
 org.glassfish.jaxb.runtime.v2.util=[3.0,4)
-org.hibernate.validator=[7.0,8)
```
Same comment as jakarta-versions-ee10.properties

mswatosh commented on dev/wlp-jakartaee-transform/rules/jakarta-versions.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen=[3.0,4)
 org.glassfish.jaxb.runtime.v2.schemagen.xmlschema=[3.0,4)
 org.glassfish.jaxb.runtime.v2.util=[3.0,4)
-org.hibernate.validator=[7.0,8)
```
Same comment as jakarta-versions-ee10.properties

mswatosh reviewed (changes requested):
Looks good, just a few minor changes and some areas to double check if we need the changes

gauthamkrishnanibm committed:
review comments changes

gauthamkrishnanibm committed:
Update dev/io.openliberty.org.hibernate.validator.9.0/bnd.overrides

Co-authored-by: Mark Swatosh <mark.swatosh@gmail.com>

gauthamkrishnanibm committed:
Update dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.1.feature

Co-authored-by: Mark Swatosh <mark.swatosh@gmail.com>

gauthamkrishnanibm committed:
Update dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/BvalManagedObjectBuilderImpl.java

Co-authored-by: Mark Swatosh <mark.swatosh@gmail.com>

gauthamkrishnanibm committed:
Update dev/io.openliberty.org.hibernate.validator.9.0/bnd.overrides

Co-authored-by: Mark Swatosh <mark.swatosh@gmail.com>

gauthamkrishnanibm committed:
Update dev/io.openliberty.org.hibernate.validator.9.0/bnd.bnd

Co-authored-by: Mark Swatosh <mark.swatosh@gmail.com>

gauthamkrishnanibm committed:
Merge branch 'integration' into upgrade-hv

gauthamkrishnanibm commented:
#build #libby

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.1.feature:
```diff
@@ -0,0 +1,14 @@
+-bundles=\
+  com.ibm.ws.beanvalidation.v20.cdi.jakarta,\
+  io.openliberty.org.hibernate.validator.cdi.9.0
+kind=ga
```
Should be beta?

tevans78 commented on dev/com.ibm.ws.beanvalidation.v20.cdi/bnd.bnd:
```diff
@@ -41,21 +42,26 @@ IBM-Default-Config: OSGI-INF/wlp/defaultInstances.xml
+	com.ibm.ws.kernel.service;version=latest,\
+	com.ibm.ws.javaee.dd;version=latest,\
+	com.ibm.ws.logging.core,\
+	com.ibm.ws.managedobject;version=latest,\
+	com.ibm.ws.org.hibernate.validator;version=latest,\
+	com.ibm.ws.org.hibernate.validator.cdi;version=latest,\
+	com.ibm.ws.kernel.feature,\
+	io.openliberty.org.hibernate.validator.cdi.9.0;version=latest,\
+	io.openliberty.org.hibernate.validator.9.0;version=latest,\
```
Building against both the old and new versions of hibernate isn't a great idea ... isn't there another way?

tevans78 commented on dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/BvalManagedObjectBuilderImpl.java:
```diff
@@ -192,10 +249,15 @@ private void createManagedConstraintValidatorFactory(Configuration<?> config, Cl
         if (constraintValidatorFactoryClassName == null) {
             // use default
-            cvf = createManagedObject(InjectingConstraintValidatorFactory.class);
+            if (isBeanValidationVersion31()) {
```
Wouldn't you be better off creating a new version of this class in a different bundle, to have a clean separation between bVal 3.0 (hibernate 7) and bVal 3.1 (hibernate 9)?

tevans78 commented on dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/BvalManagedObjectBuilderImpl.java:
```diff
@@ -285,4 +346,30 @@ private Class<?> loadClass(String className, ClassLoader appClassLoader) {
+    private void setVersion() {
+        provisionerService.getInstalledFeatures().forEach(feature -> {
+            String subString = null;
+            if (feature.startsWith("beanValidation-")) {
```
technically feature names are not case sensitive  ... you might get away with it here since you're only processing what is coming back from the installed features??

jhanders34 replied:
This is not the normal way that you should be tracking version of a feature.  This approach is definitely the anti-pattern.

tevans78 commented on dev/io.openliberty.org.hibernate.validator.9.0/bnd.bnd:
```diff
@@ -0,0 +1,20 @@
+#*******************************************************************************
+-include= jar:${fileuri;${repo;org.hibernate.validator:hibernate-validator;9.0.0.Beta3;EXACT}}!/META-INF/MANIFEST.MF,bnd.overrides
+
+-buildpath: \
```
Are you missing some stuff like this?
```
javac.source: 17
javac.target: 17

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
```

tevans78 commented on dev/io.openliberty.org.hibernate.validator.cdi.9.0/bnd.bnd:
```diff
@@ -0,0 +1,19 @@
+#*******************************************************************************
+-include= jar:${fileuri;${repo;org.hibernate.validator:hibernate-validator-cdi;9.0.0.Beta3;EXACT}}!/META-INF/MANIFEST.MF,bnd.overrides
+
+instrument.disabled: true
```
Just like the other bundle, you might want
```
javac.source: 17
javac.target: 17

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
```

tevans78 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee11.properties:
```diff
@@ -227,6 +227,7 @@ org.hibernate.validator.internal.util=[7.0,8)
 org.hibernate.validator.internal.util.classhierarchy=[7.0,8)
 org.hibernate.validator.internal.util.logging=[7.0,8)
 org.hibernate.validator.internal.util.privilegedactions=[7.0,8)
+org.hibernate.validator.internal.util.actions=[9.0,8)
```
```suggestion
org.hibernate.validator.internal.util.actions=[9.0,10)
```

gauthamkrishnanibm committed:
review comments correction

mswatosh commented:
@gauthamkrishnanibm Why were the versions reverted? Based on the pipeline results, the error is probably due to this invalid version range:
```
org.hibernate.validator.internal.util.actions=[9.0,8)
```
I would suggest removing the hibernate validator rules from the ee11 version, since ee11 should be using HV9 which doesn't need transforming

gauthamkrishnanibm commented:
@mswatosh as we discussed offline, versions files isn’t just doing the transform, it’s substituting the version ranges for the imports, so we will need this.

/com.ibm.ws.beanvalidation.v20/generated/com.ibm.ws.beanvalidation.v20.jar
                                        ```
org.hibernate.validator.internal.properties;version="[6.2,7)"
                                        org.hibernate.validator.internal.util.classhierarchy;version="[6.2,7)"
                                        org.hibernate.validator.internal.util.logging;version="[6.2,7)"
                                        org.hibernate.validator.internal.util;version="[6.2,7)"
                                        org.hibernate.validator.messageinterpolation;version="[6.2,7)"
                                        org.hibernate.validator.spi.properties;version="[6.2,7)"
                                        org.hibernate.validator;version="[6.2,7)"
```
is change in /com.ibm.ws.beanvalidation.v20/generated/com.ibm.ws.beanvalidation.v20.jakarta.jar
                                        ```
org.hibernate.validator.internal.properties;version="[7.0,10)"
                                        org.hibernate.validator.internal.util.classhierarchy;version="[7.0,10)"
                                        org.hibernate.validator.internal.util.logging;version="[7.0,10)"
                                        org.hibernate.validator.internal.util;version="[7.0,10)"
                                        org.hibernate.validator.messageinterpolation;version="[7.0,10)"
                                        org.hibernate.validator.spi.properties;version="[7.0,10)"
                                        org.hibernate.validator;version="[7.0,10)"
```

gauthamkrishnanibm committed:
fixing code review comments

gauthamkrishnanibm commented:
#build #libby

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.0.feature:
```diff
@@ -4,7 +4,7 @@ visibility=private
 IBM-Provision-Capability: \
   osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-3.0)(osgi.identity=io.openliberty.cdi-4.0)(osgi.identity=io.openliberty.cdi-4.1)))", \
-  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.beanValidation-3.0)(osgi.identity=io.openliberty.beanValidation-3.1)))"
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.beanValidation-3.0)))"
```
```suggestion
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.beanValidation-3.0))"
```

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.0.feature:
```diff
@@ -4,7 +4,7 @@ visibility=private
 IBM-App-ForceRestart: install, uninstall
 IBM-Provision-Capability: \
   osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-3.0)(osgi.identity=io.openliberty.cdi-4.0)(osgi.identity=io.openliberty.cdi-4.1)))", \
```
```suggestion
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-3.0)(osgi.identity=io.openliberty.cdi-4.0)))", \
```

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.1.feature:
```diff
@@ -0,0 +1,14 @@
+visibility=private
+IBM-App-ForceRestart: install, uninstall
+IBM-Provision-Capability: \
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-4.1)))", \
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.beanValidation-3.1)))"
```
```suggestion
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.cdi-4.1))", \
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.beanValidation-3.1))"
```

jhanders34 commented on dev/io.openliberty.org.hibernate.validator.9.0/.classpath:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" output="bin_test" path="test"/>
+	<classpathentry exported="true" kind="src" path="/com.ibm.ws.junit.extensions"/>
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.org.hibernate.validator.9.0/resources/META-INF/beans.xml:
```diff
@@ -0,0 +1,24 @@
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_2.xsd"
+   version="1.2"
+   bean-discovery-mode="all">
```
Should make this a 4.1 versioned beans.xml

jhanders34 commented on dev/io.openliberty.org.hibernate.validator.cdi.9.0/resources/META-INF/beans.xml:
```diff
@@ -0,0 +1,24 @@
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_2.xsd"
+   version="1.2"
+   bean-discovery-mode="all">
```
Again should move to version 4.1

jhanders34 commented on dev/io.openliberty.org.hibernate.validator.cdi.9.0/resources/META-INF/beans.xml:
```diff
@@ -0,0 +1,24 @@
+   bean-discovery-mode="all">
+    <scan>
+        <exclude name="org.hibernate.validator.cdi.internal.interceptor.ValidationInterceptor"/>
+<!--         <exclude name="com.ibm.ws.beanvalidation.v20.cdi.internal.LibertyValidatorFactoryBean"/> -->
+<!--         <exclude name="com.ibm.ws.beanvalidation.v20.cdi.internal.LibertyValidatorBean"/> -->
```
```suggestion
```

Any reason to leave in this commented out content?

jhanders34 commented on dev/io.openliberty.org.jboss.logging36/bnd.bnd:
```diff
@@ -0,0 +1,32 @@
+
+-include= jar:${fileuri;${repo;org.jboss.logging:jboss-logging;3.6.0.Final;EXACT}}!/META-INF/MANIFEST.MF,bnd.overrides
+
+javac.source: 17
+javac.target: 17
+
+Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
```
This doesn't match eclipse settings.  If it is meant to be Java 17+, the .classpath and org.eclipse.jdt.core.prefs files should be updated to 17 as well.

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee11.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
-org.hibernate.validator.internal.util=[7.0,8)
-org.hibernate.validator.internal.util.classhierarchy=[7.0,8)
-org.hibernate.validator.internal.util.logging=[7.0,8)
+org.hibernate.validator=[7.0,10)
+org.hibernate.validator.cdi=[7.0,10)
+org.hibernate.validator.cdi.internal=[7.0,10)
+org.hibernate.validator.internal=[7.0,10)
+org.hibernate.validator.internal.engine.valueextraction=[7.0,10)
+org.hibernate.validator.internal.properties=[7.0,10)
+org.hibernate.validator.internal.util=[7.0,10)
+org.hibernate.validator.internal.util.classhierarchy=[7.0,10)
+org.hibernate.validator.internal.util.logging=[7.0,10)
```
```suggestion
org.hibernate.validator=[9.0,10)
org.hibernate.validator.cdi=[9.0,10)
org.hibernate.validator.cdi.internal=[9.0,10)
org.hibernate.validator.internal=[9.0,10)
org.hibernate.validator.internal.engine.valueextraction=[9.0,10)
org.hibernate.validator.internal.properties=[9.0,10)
org.hibernate.validator.internal.util=[9.0,10)
org.hibernate.validator.internal.util.classhierarchy=[9.0,10)
org.hibernate.validator.internal.util.logging=[9.0,10)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee11.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.hibernate.validator.internal.util.privilegedactions=[7.0,8)
-org.hibernate.validator.messageinterpolation=[7.0,8)
-org.hibernate.validator.spi.properties=[7.0,8)
+org.hibernate.validator.messageinterpolation=[7.0,10)
+org.hibernate.validator.spi.properties=[7.0,10)
```
```suggestion
org.hibernate.validator.messageinterpolation=[9.0,10)
org.hibernate.validator.spi.properties=[9.0,10)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee10.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
-org.hibernate.validator.internal.util=[7.0,8)
-org.hibernate.validator.internal.util.classhierarchy=[7.0,8)
-org.hibernate.validator.internal.util.logging=[7.0,8)
+org.hibernate.validator=[7.0,10)
+org.hibernate.validator.cdi=[7.0,10)
+org.hibernate.validator.cdi.internal=[7.0,10)
+org.hibernate.validator.internal=[7.0,10)
+org.hibernate.validator.internal.engine.valueextraction=[7.0,10)
+org.hibernate.validator.internal.properties=[7.0,10)
+org.hibernate.validator.internal.util=[7.0,10)
+org.hibernate.validator.internal.util.classhierarchy=[7.0,10)
+org.hibernate.validator.internal.util.logging=[7.0,10)
```
```suggestion
org.hibernate.validator=[7.0,9)
org.hibernate.validator.cdi=[7.0,9)
org.hibernate.validator.cdi.internal=[7.0,9)
org.hibernate.validator.internal=[7.0,9)
org.hibernate.validator.internal.engine.valueextraction=[7.0,9)
org.hibernate.validator.internal.properties=[7.0,9)
org.hibernate.validator.internal.util=[7.0,9)
org.hibernate.validator.internal.util.classhierarchy=[7.0,9)
org.hibernate.validator.internal.util.logging=[7.0,9)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee10.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.hibernate.validator.internal.util.privilegedactions=[7.0,8)
-org.hibernate.validator.messageinterpolation=[7.0,8)
-org.hibernate.validator.spi.properties=[7.0,8)
+org.hibernate.validator.messageinterpolation=[7.0,10)
+org.hibernate.validator.spi.properties=[7.0,10)
```
```suggestion
org.hibernate.validator.messageinterpolation=[7.0,9)
org.hibernate.validator.spi.properties=[7.0,9)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
-org.hibernate.validator.internal.util=[7.0,8)
-org.hibernate.validator.internal.util.classhierarchy=[7.0,8)
-org.hibernate.validator.internal.util.logging=[7.0,8)
+org.hibernate.validator=[7.0,10)
+org.hibernate.validator.cdi=[7.0,10)
+org.hibernate.validator.cdi.internal=[7.0,10)
+org.hibernate.validator.internal=[7.0,10)
+org.hibernate.validator.internal.engine.valueextraction=[7.0,10)
+org.hibernate.validator.internal.properties=[7.0,10)
+org.hibernate.validator.internal.util=[7.0,10)
+org.hibernate.validator.internal.util.classhierarchy=[7.0,10)
+org.hibernate.validator.internal.util.logging=[7.0,10)
```
```suggestion
org.hibernate.validator=[7.0,9)
org.hibernate.validator.cdi=[7.0,9)
org.hibernate.validator.cdi.internal=[7.0,9)
org.hibernate.validator.internal=[7.0,9)
org.hibernate.validator.internal.engine.valueextraction=[7.0,9)
org.hibernate.validator.internal.properties=[7.0,9)
org.hibernate.validator.internal.util=[7.0,9)
org.hibernate.validator.internal.util.classhierarchy=[7.0,9)
org.hibernate.validator.internal.util.logging=[7.0,9)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.hibernate.validator.internal.util.privilegedactions=[7.0,8)
-org.hibernate.validator.messageinterpolation=[7.0,8)
-org.hibernate.validator.spi.properties=[7.0,8)
+org.hibernate.validator.messageinterpolation=[7.0,10)
+org.hibernate.validator.spi.properties=[7.0,10)
```
```suggestion
org.hibernate.validator.messageinterpolation=[7.0,9)
org.hibernate.validator.spi.properties=[7.0,9)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9and10.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
-org.hibernate.validator.internal.util=[7.0,8)
-org.hibernate.validator.internal.util.classhierarchy=[7.0,8)
-org.hibernate.validator.internal.util.logging=[7.0,8)
+org.hibernate.validator=[7.0,10)
+org.hibernate.validator.cdi=[7.0,10)
+org.hibernate.validator.cdi.internal=[7.0,10)
+org.hibernate.validator.internal=[7.0,10)
+org.hibernate.validator.internal.engine.valueextraction=[7.0,10)
+org.hibernate.validator.internal.properties=[7.0,10)
+org.hibernate.validator.internal.util=[7.0,10)
+org.hibernate.validator.internal.util.classhierarchy=[7.0,10)
+org.hibernate.validator.internal.util.logging=[7.0,10)
```
```suggestion
org.hibernate.validator=[7.0,9)
org.hibernate.validator.cdi=[7.0,9)
org.hibernate.validator.cdi.internal=[7.0,9)
org.hibernate.validator.internal=[7.0,9)
org.hibernate.validator.internal.engine.valueextraction=[7.0,9)
org.hibernate.validator.internal.properties=[7.0,9)
org.hibernate.validator.internal.util=[7.0,9)
org.hibernate.validator.internal.util.classhierarchy=[7.0,9)
org.hibernate.validator.internal.util.logging=[7.0,9)
```

jhanders34 commented on dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9and10.properties:
```diff
@@ -217,18 +217,18 @@ org.glassfish.jaxb.runtime.v2.runtime.unmarshaller=[3.0,4)
 org.hibernate.validator.internal.util.privilegedactions=[7.0,8)
-org.hibernate.validator.messageinterpolation=[7.0,8)
-org.hibernate.validator.spi.properties=[7.0,8)
+org.hibernate.validator.messageinterpolation=[7.0,10)
+org.hibernate.validator.spi.properties=[7.0,10)
```
```suggestion
org.hibernate.validator.messageinterpolation=[7.0,9)
org.hibernate.validator.spi.properties=[7.0,9)
```

gauthamkrishnanibm committed:
changes for Jared's code review comments

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9and10.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9and10.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee9.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee10.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee10.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee11.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/wlp-jakartaee-transform/rules/jakarta-versions-ee11.properties

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/io.openliberty.org.hibernate.validator.9.0/.classpath

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.1.feature

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.0.feature

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.beanValidationCDI-3.0.feature

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=54f53d44-021c-410b-aecb-604491cc5e58

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 19 infrastructure code files were changed.

- 45 product code files were changed.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 19 infrastructure code files were changed.

- 45 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=385e8c03-ac51-47d7-b4bf-4172f7ca017f completed successfully!

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=54f53d44-021c-410b-aecb-604491cc5e58 has errors/failures.

jhanders34 commented on dev/io.openliberty.org.hibernate.validator.9.0/resources/META-INF/beans.xml:
```diff
@@ -0,0 +1,24 @@
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee"
```
```suggestion
<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
```

jhanders34 commented on dev/io.openliberty.org.hibernate.validator.cdi.9.0/resources/META-INF/beans.xml:
```diff
@@ -0,0 +1,22 @@
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee"
```
```suggestion
<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
```

gauthamkrishnanibm committed:
Update dev/io.openliberty.org.hibernate.validator.9.0/resources/META-INF/beans.xml

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
Update dev/io.openliberty.org.hibernate.validator.cdi.9.0/resources/META-INF/beans.xml

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

gauthamkrishnanibm committed:
some cleaning up as suggested in code review

updating the injectValidatorFactoryResources method to take a RuntimeVersion argument and passing that argument to createManagedConstraintValidatorFactory and just use it instead of setting it again in BvalManagedObjectBuilderImpl

gauthamkrishnanibm commented:
#build #libby

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=289b8aba-6ffb-4df8-be91-57b76f362cb9

Target locations of links might be accessible only to IBM employees.

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=5335f189-7891-45a5-b949-f35b1e32cdfe

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 19 infrastructure code files were changed.

- 47 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=289b8aba-6ffb-4df8-be91-57b76f362cb9 completed successfully!

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=5335f189-7891-45a5-b949-f35b1e32cdfe has errors/failures.

gauthamkrishnanibm commented:
The pipeline result looks good. The error/failures are triaged ones.
